subprojects/core/src/integTest/groovy/org/gradle/process/internal/CancellationBuildOperationIntegrationTest.groovy

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.process.internal
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.BuildOperationsFixture
+import org.gradle.test.precondition.Requires
+import org.gradle.test.preconditions.IntegTestPreconditions
+
+import static org.hamcrest.CoreMatchers.anyOf
+import static org.hamcrest.CoreMatchers.equalTo
+
+class CancellationBuildOperationIntegrationTest extends AbstractIntegrationSpec {
+
+    BuildOperationsFixture operations = new BuildOperationsFixture(executer, temporaryFolder)
+
+    @Requires(IntegTestPreconditions.NotEmbeddedExecutor)
+    def "task operations are closed even when interrupting the execution workers"() {
+        def parallelTaskCount = 5
+        buildFile << """
+          ext.workerThreads = new HashSet<>()
+          ext.latch = new java.util.concurrent.CountDownLatch(${parallelTaskCount} )
+          tasks.register('interrupting') {
+            doFirst {
+              rootProject.ext.latch.await()
+              rootProject.ext.workerThreads.each {
+                if (it.name.contains('included builds')) {
```

Comment:
This is a bit fragile. If we refactor the scheduling, it would be easy to break this condition and silently no longer interrupt anything.

Reply:
I agree. What do you suggest I do instead? We can go back to the drawing board and try to reproduce the problem in a better way.

Reply:
I am checking now the stack trace of the Thread to decide whether to kill it or not.